src/astro/databases/snowflake.py

```diff
@@ -57,6 +76,238 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
         qualified_name = ".".join(name for name in qualified_name_lists if name)
         return qualified_name
 
+    def check_optimised_path_and_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> bool:
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+        """
+        method_name = self.native_support.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            try:
+                if transfer_method(
+                    source_file=source_file,
+                    target_table=target_table,
+                    chunk_size=chunk_size,
+                    if_exists=if_exists,
+                    **kwargs,
+                ):
+                    return True
+            except Exception as exec_err:
+                raise exec_err
+        return False
+
+    @staticmethod
+    def get_gcs_project_id_from_conn(conn: connection) -> str:
+        """
+        Get GCS project id from conn
+        :param conn: Airflow's connection
+        """
+        if conn.extra and conn.extra.get("project"):
+            return str(conn.extra.get("project"))
+        elif conn.host:
+            return str(conn.host)
+        raise ValueError(f"conn_id {conn.conn_id} has no project id.")
+
+    def gs_to_snowflake(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> bool:
+        """
+        Checks if the table and schema exists on Snowflake.
+        If table and schema exists, create stage based on integration passed by user.
+        Generate query and run COPY INTO command and return True else returns False
+        @param source_file: Source file
+        @param target_table: Output target table on snowflake
+        @param chunk_size: Chunk size for the file
+        @param if_exists: Update strategy for file
+        @param kwargs:
+        @return:bool
```

Comment:
```suggestion
        Generate query and run COPY INTO command and return True else returns False

        :param source_file: Source file
        :param target_table: Output target table on snowflake
        :param chunk_size: Chunk size for the file
        :param if_exists: Update strategy for file
        :param kwargs:
        :return: bool
```

`:return:` should says what is being returned, `:rtype` shows the return type